### PR TITLE
use base64/uudecode if present to validate tar xz support

### LINKED ISSUE #54: prefer `.tar.xz` downloads if the system tar can decompress them
The tricky part here is determining if `tar` supports this format.

On mac the answer is easy: yes.

On other platforms, we need a test.

> We want to prefer `.tar.xz` so we download faster, tea is unfortunately large due to `deno` and this is not ideal for users with less available bandwidth

### PR DESCRIPTION
**Pull request by jhheider**
resolves #54

### CODE REVIEW

**Comment by mxcl** [comment]
This is one of the more epic PRs I've ever seen.

**Comment by mxcl** [comment]
Thwarted because some bottles don’t have `.tar.xz` available.

Can we fix that?

**Comment by jhheider** [comment]
almost certainly everything prior to the rebuild is going to be removed. I can write a script to move it out of the way for now.

**Comment by jhheider** [comment]
Fixed unicode at least (we somehow got a 71.1.1 in there, but 71.1.0 is the latest in the pantry, so that's what we're building). the longer cleanup is more complicated, so I'll think about how to cleanly script that, since we need to update inventories as well.

**Comment by jhheider** [comment]
added a few fixes:
- [x] some shellcheck bashism from my work basing on the recent merge
- [x] clean up accidental output
- [x] pass xJ instead of xz for xz vs. gz (this is confusing but real)
- [x] clean up one of the CI checks to deal with the fact that `v0.10` doesn't match shell pattern `v0.?`

File: install.sh [hunk 1]
```diff
@@ -29,17 +31,47 @@ prepare() {
...
+				EOF
+			if uudecode -p "$TMPFILE" | tar Jtf - >/dev/null 2>&1; then
+				ZZ=xz
+			fi
+		fi
+	fi
+
+	case "$ZZ" in
+	gz)
+		TAR_FLAGS=xz # confusingly
```
Comment (mxcl) [inline install.sh:68]: lol indeed

**Review by mxcl — APPROVED** [review]

**Pull request merged**